src/views/catalog/wizard/tabs/environment/WizardEnvironmentTab.tsx

```diff
@@ -1,107 +1,28 @@
+import { Bullseye } from '@patternfly/react-core';
 
 const WizardEnvironmentTab: WizardTab = ({ vm, updateVM, setDisableVmCreate }) => {
-  const { ns } = useParams<{ ns: string }>();
-  const { t } = useKubevirtTranslation();
-
-  const {
-    secrets,
-    configMaps,
-    serviceAccounts,
-    loaded,
-    error: loadError,
-  } = useEnvironmentsResources(ns);
-
-  const {
-    environments,
-    onEnvironmentAdd,
-    onEnvironmentChange,
-    onEnvironmentRemove,
-    onReload,
-    onSave,
-    edited,
-    error: formError,
-    setError,
-  } = useEnvironments(vm, updateVM);
-
-  React.useEffect(() => {
-    setDisableVmCreate(edited);
-    return () => setDisableVmCreate(false);
-  }, [edited, setDisableVmCreate]);
-
-  if (!loaded) return <WizardEnvironmentTabSkeleton />;
-
-  const environmentNamesSelected = environments.map((env) => env.name);
+  const onEditChange = React.useCallback(
+    (edited) => {
+      setDisableVmCreate(edited);
+    },
+    [setDisableVmCreate],
+  );
 
-  const closeError = () => {
-    setError(undefined);
-  };
+  if (!vm)
+    return (
+      <Bullseye>
+        <Loading />
+      </Bullseye>
+    );
 
   return (
     <div className="co-m-pane__body">
-      <WizardEnvironmentTabTitle />
-
-      {environments.length !== 0 && (
-        <div className="row pairs-list__heading">
-          <div className="col-xs-5 text-secondary text-uppercase" id="environment-name-header">
-            {t('config map / secret / service account')}
-          </div>
-          <div className="col-xs-5 text-secondary text-uppercase" id="environment-serial-header">
-            {t('Serial Number')}
-          </div>
-          <div className="col-xs-1 co-empty__header" />
-        </div>
-      )}
-
-      {environments.map((environment, index) => (
-        <EnvironmentEditor
-          key={environment.name}
-          environmentName={environment.name}
-          serial={environment?.serial}
-          kind={environment.kind}
-          secrets={secrets}
-          configMaps={configMaps}
-          serviceAccounts={serviceAccounts}
-          onChange={onEnvironmentChange}
-          onRemove={onEnvironmentRemove}
-          id={index}
-          environmentNamesSelected={environmentNamesSelected}
-        />
-      ))}
-
-      <div className="row">
-        <div className="col-xs-12">
-          <Button
-            className="pf-m-link--align-left"
-            onClick={onEnvironmentAdd}
-            type="button"
-            variant="link"
-          >
-            <PlusCircleIcon /> {t('Add Config Map, Secret or Service Account')}
-          </Button>
-        </div>
-      </div>
-
-      <WizardEnvironmentTabFooter
-        error={loadError || formError}
-        onSave={onSave}
-        onReload={onReload}
-        closeError={closeError}
-        isSaveDisabled={!edited || !environments.every((env) => env.name)}
-      />
+      <EnvironmentForm vm={vm} onEditChange={onEditChange} updateVM={updateVM} />
```

Comment:
i think if you are using once you can simply use (edited) => setDisableVmCreate(edited)

Reply:
i can event pass only `setDisableVMCreate` :-) Thanks for the suggestion